### PR TITLE
Add Door Event

### PR DESCRIPTION
**Pull request by Rysik5318**

### CODE REVIEW

File: Exiled.Events/EventArgs/Door/DoorDamagingEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        /// </summary>
+        public bool IsAllowed { get; set; }
+
+        /// <summary>
+        ///    Gets or sets a value indicating the type of door damage (DoorDamageType)
+        /// </summary>
+        public DoorDamageType DoorDamageType { get; set; }
+
+        /// <summary>
+        ///     Gets or sets a value indicating the damage received by the door
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/EventArgs/Door/DoorDamagingEventArgs.cs:50]: Docs in here is confusing, is it supposed to be the remaining HP of the door, or the damage amount?

**Review by louis1706 — CHANGES_REQUESTED** [review]
image
DamagingDoor event already exist

**Comment by Rysik5318** [comment]
> image DamagingDoor event already exist

Should we move this event to `Door'?

**Comment by VALERA771** [comment]
> > image DamagingDoor event already exist
> 
> 
> 
> Should we move this event to `Door'?

That event patches the same method so I think you need to merge your events with that patch

**Comment by louis1706** [comment]
> > image DamagingDoor event already exist
> 
> Should we move this event to `Door'?

don't make breaking change

File: Exiled.Events/EventArgs/Door/DoorDamagingEventArgs.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        /// </summary>
+        public float Hp { get; set; }
```
Comment (louis1706) [inline Exiled.Events/EventArgs/Door/DoorDamagingEventArgs.cs:52]: Health

**Review by louis1706 — CHANGES_REQUESTED** [review]
since there BreakingChange this PR will need to wait Exiled 9 (or remove breaking change)

**Comment by Rysik5318** [comment]
I'll wait for Exiled 9.

**Comment by louis1706** [comment]
> I'll wait for Exiled 9.

ok i would love than next update they would use DamageHandler with Door lol would be more logic xD

**Comment by Rysik5318** [comment]
xD

**Comment by louis1706** [comment]
I have fixed all the conflict

File: Exiled.Events/Patches/Events/Map/ServerDamage.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            __instance.RemainingHealth -= Health;
+            if (__instance.RemainingHealth <= 0f)
+            {
+                DestroyingDoorEventArgs destroyingDoorEventArgs = new(__instance);
+
+                Handlers.Map.OnDoorDestroying(destroyingDoorEventArgs);
+
+                if (!destroyingDoorEventArgs.IsAllowed)
+                {
+                    __instance.RemainingHealth += Health;
```
Comment (louis1706) [inline Exiled.Events/Patches/Events/Map/ServerDamage.cs:65]:
```suggestion
```

File: Exiled.Events/EventArgs/Map/DestroyedDoorEventArgs.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    /// </summary>
+    public class DestroyedDoorEventArgs : IDoorEvent
+    {
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="DestroyedDoorEventArgs" /> class.
+        /// </summary>
+        /// <param name="doorVariant">
+        ///     <inheritdoc cref="DoorVariant" />
+        /// </param>
+        public DestroyedDoorEventArgs(DoorVariant doorVariant)
```
Comment (Misfiy) [inline Exiled.Events/EventArgs/Map/DestroyedDoorEventArgs.cs:24]: Add DoorDamageType here too

File: Exiled.Events/EventArgs/Map/DestroyingDoorEventArgs.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="DestroyingDoorEventArgs" /> class.
+        /// </summary>
+        /// <param name="doorVariant">
+        ///     <inheritdoc cref="DoorVariant" />
+        /// </param>
+        /// <param name="isAllowed">
+        ///     <inheritdoc cref="IsAllowed" />
+        /// </param>
+        public DestroyingDoorEventArgs(DoorVariant doorVariant, bool isAllowed = true)
```
Comment (Misfiy) [inline Exiled.Events/EventArgs/Map/DestroyingDoorEventArgs.cs:27]: Add DoorDamageType here too

**Pull request merged**